### PR TITLE
Move registerView from SdkMeterProvider to SdkMeterProviderBuilder. (#3060)

### PR DESCRIPTION
**Pull request by jack-berg**
Clears the way for #3060 by making SdkMeterProvider immutable.

### CODE REVIEW

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeterProvider.java [hunk 1]
```diff
@@ -79,30 +83,4 @@ public SdkMeter get(String instrumentationName, @Nullable String instrumentation
...
-   *
-   * // create a specification of how you want the metrics aggregated:
-   * AggregatorFactory aggregatorFactory = AggregatorFactory.minMaxSumCount();
-   *
-   * //register the view with the MeterSdkProvider
-   * meterProvider.registerView(instrumentSelector, View.builder()
-   *   .setAggregatorFactory(aggregatorFactory).build());
-   * }</pre>
-   */
-  public void registerView(InstrumentSelector selector, View view) {
```
Comment (jkwatson) [inline sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeterProvider.java:105]: let's keep this, but deprecate it, and remove it in the next release, to provide a smoother transition for people who might already be using the existing views API directly.

Comment (jack-berg) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeterProvider.java:105]: 👌

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeterProviderBuilder.java [hunk 2]
```diff
@@ -47,6 +52,35 @@ public SdkMeterProviderBuilder setResource(@Nonnull Resource resource) {
...
+   * <p>Example on how to register a view:
+   *
+   * <pre>{@code
+   * // create a SdkMeterProviderBuilder
+   * SdkMeterProviderBuilder meterProviderBuilder = SdkMeterProvider.builder();
+   *
+   * // create a selector to select which instruments to customize:
+   * InstrumentSelector instrumentSelector = InstrumentSelector.builder()
+   *   .setInstrumentType(InstrumentType.COUNTER)
+   *   .buildInstrument();
```
Comment (jkwatson) [inline sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeterProviderBuilder.java:67]:
```suggestion
   *   .build();
```

**Review by jkwatson — APPROVED** [review]
Looks good. Just one little javadoc cleanup, then 👍

**Comment by jack-berg** [comment]
Great! Appreciate the quick feedback! Your javadoc comment made me take a second look at the view registry code sample and I found some other typos.

**Review by anuraaga — APPROVED** [review]

**Pull request merged**